GeosCore/fullchem_mod.F90

```diff
@@ -1130,6 +1132,30 @@ SUBROUTINE Do_FullChem( Input_Opt,  State_Chm, State_Diag,                 &
 #endif
        ENDIF
 
+#if defined( MODEL_GEOS )
+       ! Check for negative concentrations after first integration and 
+       ! eventually mark integration as errorenous so that it will be
+       ! repeated below. cakelle2, 2023/10/26.
+       IF ( Input_Opt%KppCheckNegatives >= 0 ) THEN
+          IF ( ( Input_Opt%KppCheckNegatives==0 .AND. State_Met%InStratMeso(I,J,L) ) .OR. &
+               ( L > (State_Grid%NZ-Input_Opt%KppCheckNegatives) ) ) THEN
+             IF ( ANY(C < 0.0_dp) ) THEN
+                IERR = -999
+                IF ( State_Diag%Archive_KppError ) THEN
+                   State_Diag%KppError(I,J,L) = State_Diag%KppError(I,J,L) + 1.0 
```

Comment:
The increment should be `1.0_f4` since I believe KppError is a single-precision field.

Reply:
Done.